libeos-payg/manager.c

```diff
@@ -647,7 +652,7 @@ set_expiry_time (EpgManager *self,
     {
       g_assert (self->expiry_time_secs >= now_secs);
       g_assert (self->expiry_time_secs - now_secs <= G_MAXUINT);
-      self->expiry = g_timeout_source_new_seconds (MIN (self->expiry_time_secs - now_secs, 60));
+      self->expiry = epg_clock_source_new_seconds (self->clock, MIN (self->expiry_time_secs - now_secs, 60));
```

Comment:
I think the point of ticking the clock every 60 seconds was to ensure that changes to the wall-clock time were noticed within 60 seconds? So in principle this could actually just be set to seconds-to-expiry.

Reply:
Yes I think you're right